app/main.py

```diff
@@ -1,2 +1,121 @@
+from typing import Any, Iterable
+
+
+class Node:
+    def __init__(self, key: Any, value: Any, hash_value: int) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash_value
+
+
 class Dictionary:
-    pass
+    DELETED = "DELETED"
+
+    def __init__(self) -> None:
+        """Initialize the Dictionary with default capacity and load factor."""
+        self.capacity = 8
+        self.length = 0
+        self.load_factor = 2 / 3
+        self.hash_table = [None] * self.capacity
+
+    def _hash(self, key: Any) -> int:
+        """Return the hash index for a given key."""
+        return hash(key) % self.capacity
+
+    def _resize(self) -> None:
```

Comment:
The `pop` method checks for a default value but doesn’t address the case where the default is `None`. It might be clearer to return the default only when the key doesn’t exist.